src/Toolchain/ToolchainProvider.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as vscode from 'vscode';
+import {gToolchainEnvMap} from './ToolchainEnv';
+
+enum NodeType {
+  backend,
+  toolchain,
+}
+
+export class ToolchainNode extends vscode.TreeItem {
+  constructor(
+      public readonly label: string,
+      public readonly collapsibleState: vscode.TreeItemCollapsibleState,
+      public readonly type: NodeType,
+      public readonly version?: string,
+  ) {
+    super(label, collapsibleState);
+
+    if (type === NodeType.backend) {
+      this.iconPath = new vscode.ThemeIcon('bracket');
+    } else if (type === NodeType.toolchain) {
```

Comment:
(optional) "ToolchainNode type can be of toolchain or _backend_" could be confusing. 
How about renaming `ToolchainNode` to `ToolchainViewNode` ?

Reply:
This variable always uses with `NodeType`. So I think it seems unnecessary that the node is attached to the name again.